hook.go

```diff
@@ -62,84 +67,51 @@ func queryString(query interface{}) ([]byte, error) {
 	}
 }
 
-type queryProcessedHook func(*QueryProcessedEvent)
-
-// OnQueryProcessed calls the fn with QueryProcessedEvent
+// OnQueryEvent calls the interface with QueryEvent
 // when query is processed.
-func (db *DB) OnQueryProcessed(fn func(*QueryProcessedEvent)) {
-	db.queryProcessedHooks = append(db.queryProcessedHooks, fn)
+func (db *DB) OnQueryEvent(hook Hook) {
```

Comment:
I will change this myself, but I think this should be `AddHook` or `AddQueryHook`. I expect `OnSomething` to receive a function.

Reply:
Yeah, I didn’t give the name much thought, I agree.